### PR TITLE
London Class 7 - Rafael Flores - JavaScript Core 1 - Week 3

### PR DESCRIPTION
**Pull request by rfloresram**
# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

### CODE REVIEW

File: 2-mandatory/4-eligible-students.js [hunk 1]
```diff
@@ -7,8 +7,10 @@
   - Returns an array containing only the names of the who have attended AT LEAST 8 classes
 */
 
-function getEligibleStudents() {
-
+function onlyStudents(array, i) {
```
Comment (bisioolajide) [inline 2-mandatory/4-eligible-students.js:10]: Great work Rafael! I like your approach to this and got some tips from you.

File: 1-exercises/D-array-filter/exercise.js [hunk 2]
```diff
@@ -6,9 +6,12 @@
   - Do not edit any of the existing code
 */
 
-var pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1], false, "whoops"];
+var pairingArr = [[0, 3], [1, 2], [2, 1], null, [1], false, "Nope!"];
```
Comment (lpalma) [inline 1-exercises/D-array-filter/exercise.js:9]: Well done Rafael :clap: 
But let's remember that the exercise asks us not to edit any of the existing code :)

File: 1-exercises/E-array-map/exercise.js [hunk 3]
```diff
@@ -2,3 +2,14 @@
...
 var numbers = [0.1, 0.2, 0.3, 0.4, 0.5];
+
+/* let multiplyNumbersBy100 = numbers.map(num => num * 100);*/
+
+function multiplyNum (numb) {
+    return numb * 100;
+ }
+
+ var multipledNum = numbers.map(multiplyNum);
+ console.log(multipledNum);
```
Comment (lpalma) [inline 1-exercises/E-array-map/exercise.js:13]: Nice :+1: 
Are there other possible solutions, using different syntax, as asked in the exercise description?

File: 1-exercises/H-array-methods-2/exercise2.js [hunk 4]
```diff
@@ -7,7 +7,9 @@
   Tip: use the string method .split() and the array method .join()
 */
 
-function capitalise(str) {}
+function capitalise(str) {
+  return str.charAt(0).toUpperCase() + str.slice(1)
+}
```
Comment (lpalma) [inline 1-exercises/H-array-methods-2/exercise2.js:12]: hi Rafael,
Although we are capitalising the string, the exercise asks us to use the function `.join()` as part of the solution.
You can find more information about how to use the `.join` function here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join

**Review by lpalma — COMMENTED** [review]
Good work Rafael :+1: 
I couldn't find exercises **I** and **J**. Keep up with the good work :clap:

File: 2-mandatory/1-oxygen-levels.js [hunk 5]
```diff
@@ -11,7 +11,9 @@
     Some string methods that might help you here are .replace() and .substring(). 
 */
 
-function findSafeOxygenLevel() {}
+function safeLevels(arr) {
+    const safeLevels = (arr) => arr.map(level => level.substring(0, 4)).find(num => num > 19.5 && num < 23.5) + "%"
```
Comment (lpalma) [inline 2-mandatory/1-oxygen-levels.js:15]: hi Rafael :wave: 
Your solution is very close. Seems like you are assigning a `const` but never returning any value from your `safeLevels` function

File: 2-mandatory/2-bush-berries.js [hunk 6]
```diff
@@ -44,13 +50,18 @@ function test(test_name, actual, expected) {
...
+  "functionElement function works - case 1",
+  functionElement(bushBerryColours1),
   "Toxic! Leave bush alone!"
 );
 
 test(
-  "isBushSafe funtion works - case 1",
-  isBushSafe(bushBerryColours2),
+  "functionElement function works - case 1",
+  functionElement(bushBerryColours2),
```
Comment (lpalma) [inline 2-mandatory/2-bush-berries.js:60]: On line **32** we have a comment asking us not to modify the test. Even a simple change of name is considered a modification and we should avoid it :)

File: 2-mandatory/3-space-colonies.js [hunk 7]
```diff
@@ -49,8 +61,10 @@ function test(test_name, actual, expected) {
   console.log(`${test_name}: ${status}`);
 }
 
-test("getSettlers function works", getSettlers(voyagers), [
+test("colonizerElement function works", colonizerElement(voyagers), [
```
Comment (lpalma) [inline 2-mandatory/3-space-colonies.js:64]: once again please let's not modify any code which we are not supposed to :)

**Review by lpalma — COMMENTED** [review]
Getting there Rafael :clap:

**Pull request closed without merging**